registration/api/v2/views.py

```diff
@@ -0,0 +1,219 @@
+        responses={**VALIDATION_RESPONSE, 200: openapi.Response("Validation Successful")},
+        tags=["v2", "Instances", "OpenEdXInstanceConfig"],
+        security=[],
+    )
+    @action(detail=False, methods=['post'])
+    def validate(self, request):
+        """
+        Validate instance configuration
+
+        This action is publicly accessible and allows any user to validate an instance
+        configuration. It is useful when signing up.
+        """
+        serializer = self.get_serializer(data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        return Response(status=status.HTTP_200_OK)
+
+    @swagger_auto_schema(
+        responses={**VALIDATION_AND_AUTH_RESPONSES, 200: openapi.Response("Changes committed")},
+        tags=["v2", "Instances", "OpenEdXInstanceConfig"],
+        manual_parameters=[
+            openapi.Parameter(
+                'force',
+                openapi.IN_QUERY,
+                type=openapi.TYPE_BOOLEAN,
+                description="Force launching a new instance even if one is already in progress.",
+            )
+        ]
+    )
+    @action(detail=True, methods=['post'])
+    def commit_changes(self, request, pk=None):
+        """
+        Commit configuration changes to instance and launch new AppServer.
+
+        This API call will copy over any changes made to the instance config to
+        the actual instance used to launch AppServers and launch a new AppServer
+        with the applied changes.
+
+        It checks to see if an AppServer is already being provisioned and in that
+        case prevents a new one from being launched unless forced.
+        """
+        force = request.query_params.get('force', False)
+        instance_config: BetaTestApplication = self.get_object()
+        instance = instance_config.instance
+        if instance is None:
+            # For a new user an instance will not exist until they've verified their email.
+            raise ValidationError("Must verify email before launching an instance", code="email-unverified")
+        in_progress_statuses = Status.New.state_id, Status.ConfiguringServer.state_id, Status.WaitingForServer.state_id
+
+        if not force and instance.appserver_set.filter(_status__in=in_progress_statuses).exists():
+            raise ValidationError("Instance launch already in progress", code="in-progress")
+
+        if not EmailAddress.objects.get(email=instance_config.public_contact_email).is_confirmed:
+            raise ValidationError("Updated public email needs to be confirmed.", code="email-unverified")
+
+        instance.theme_config = instance_config.draft_theme_config
+        instance.name = instance_config.instance_name
+        instance.privacy_policy_url = instance_config.privacy_policy_url
+        instance.email = instance_config.public_contact_email
+        instance.save()
+
+        spawn_appserver(instance.ref.pk, mark_active_on_success=True, num_attempts=2)
```

Comment:
I believe it might be nice to extract `num_attempts` to settings and env variables instead of hardcoding it.